### PR TITLE
Chest/EnderChest Sounds and Animations

### PR DESCRIPTION
**Pull request by wow4201**
Plays the animations and sounds for chest and enderchest. Tested and working. Squashed.

### CODE REVIEW

File: src/main/java/net/glowstone/block/blocktype/BlockChest.java [hunk 1]
```diff
@@ -33,4 +39,22 @@ public void placeBlock(GlowPlayer player, GlowBlockState state, BlockFace face,
...
+		if (((GlowChest) block.getState()).getState() == 0){
+			block.getWorld().playSound(block.getLocation(), Sound.CHEST_OPEN, 1, 1);
+			player.SetBindChest(block.getLocation());
+			((GlowChest) block.getState()).setState((byte)1);
+			((GlowChest) block.getState()).ChestAnimation((byte)1);
+		}
+		return true;
+	}
+	return false;
+	}
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/blocktype/BlockChest.java:58]: Indentation

File: src/main/java/net/glowstone/block/blocktype/BlockChest.java [hunk 2]
```diff
@@ -33,4 +39,22 @@ public void placeBlock(GlowPlayer player, GlowBlockState state, BlockFace face,
             GlowServer.logger.warning("Placing Chest: MaterialData was of wrong type");
         }
     }
+    
+    @Override
+    public boolean blockInteract(GlowPlayer player, GlowBlock block, BlockFace face, Vector clickedLoc) {
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/blocktype/BlockChest.java:44]: I spy tabs in this entire method

File: src/main/java/net/glowstone/block/blocktype/BlockEnderchest.java [hunk 3]
```diff
@@ -15,12 +22,28 @@
...
+        return new TEEnderChest(chunk.getBlock(cx, cy, cz));
+    }
 
     @Override
     public boolean blockInteract(GlowPlayer player, GlowBlock block, BlockFace face, Vector clickedLoc) {
-        // todo: animation?
-        player.openInventory(player.getEnderChest());
-        return true;
+	//player.openInventory(player.getEnderChest());
+	TileEntity te = block.getTileEntity();
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/blocktype/BlockEnderchest.java:34]: Tabs below here (and here)

File: src/main/java/net/glowstone/block/state/GlowChest.java [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
+import net.glowstone.entity.GlowPlayer;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
 import org.bukkit.block.Chest;
 import org.bukkit.inventory.Inventory;
 
 public class GlowChest extends GlowBlockState implements Chest {
 
+	public int state;
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/state/GlowChest.java:16]: Tab

edit: and why public

File: src/main/java/net/glowstone/block/state/GlowChest.java [hunk 5]
```diff
@@ -26,4 +33,28 @@ public Inventory getInventory() {
         // todo: handle double chests
         return getBlockInventory();
     }
+    
+    //closed 0, open 1
+    public int getState() {
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/state/GlowChest.java:38]: Maybe an enum would be better for open/closed? What if there is a tri-state in the future?

File: src/main/java/net/glowstone/block/state/GlowEnderChest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import net.glowstone.entity.GlowPlayer;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Chest;
+import org.bukkit.inventory.Inventory;
+
+public class GlowEnderChest extends GlowBlockState implements Chest {
+
+	public int state;
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/state/GlowEnderChest.java:16]: Tabs. Also why is this public.

File: src/main/java/net/glowstone/block/state/GlowEnderChest.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        return getTileEntity().getInventory();
+    }
+
+    @Override
+    public Inventory getInventory() {
+        // todo: handle double chests
+        return getBlockInventory();
+    }
+    
+    //closed 0, open 1
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/state/GlowEnderChest.java:37]: https://github.com/SpaceManiac/Glowstone/pull/211/files#r17217994

File: src/main/java/net/glowstone/block/state/GlowEnderChest.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    public Inventory getBlockInventory() {
+        return getTileEntity().getInventory();
+    }
+
+    @Override
+    public Inventory getInventory() {
+        // todo: handle double chests
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/state/GlowEnderChest.java:33]: Might want to handle this before it gets pulled ;P

File: src/main/java/net/glowstone/entity/GlowPlayer.java [hunk 9]
```diff
@@ -232,6 +232,12 @@
      * The player's base walking speed.
      */
     private float walkSpeed = 0.2f;
+    
+    /**
+     * If the player currently has an open chest.
```
Comment (turt2live) [inline src/main/java/net/glowstone/entity/GlowPlayer.java:237]: But this isn't a boolean?

File: src/main/java/net/glowstone/net/handler/play/inv/CloseWindowHandler.java [hunk 10]
```diff
@@ -21,5 +30,24 @@ public void handle(GlowSession session, CloseWindowMessage message) {
             }
             player.setItemOnCursor(null);
         }
+        
+        if (player.GetBindChest() != null){
```
Comment (turt2live) [inline src/main/java/net/glowstone/net/handler/play/inv/CloseWindowHandler.java:34]: Indentation issues below here.

File: src/main/java/net/glowstone/block/entity/TEEnderChest.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+import net.glowstone.block.GlowBlock;
+import net.glowstone.block.GlowBlockState;
+import net.glowstone.block.state.GlowEnderChest;
+import org.bukkit.event.inventory.InventoryType;
+
+/**
+ * Tile entity for Chests.
+ */
+public class TEEnderChest extends TEContainer {
```
Comment (SpaceManiac) [inline src/main/java/net/glowstone/block/entity/TEEnderChest.java:11]: I don't think Ender chests actually have tile entities.

Comment (turt2live) [inline src/main/java/net/glowstone/block/entity/TEEnderChest.java:11]: 90% sure it's just a chest.

**Comment by turt2live** [comment]
@wow4201 Your PR is currently not mergable. Please fix this for further review. Thanks!

**Comment by dequis** [comment]
Simple conflicts in BlockChest.java and BlockEnderchest.java, newline changes in GlowPlayer.java and a simple conflict behind it. Use `git merge -X renormalize` to handle the newline changes. Result of my merge here https://github.com/dequis/Glowstone/commit/f9401b1ef3830b6bcb844f238632999bee994eb8

File: src/main/java/net/glowstone/block/state/GlowEnderChest.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+        state = 0;
+    }
+
+    private TEEnderChest getTileEntity() {
+        return (TEEnderChest) getBlock().getTileEntity();
+    }
+
+    @Override
+    public Inventory getBlockInventory() {
+        return getTileEntity().getInventory();
```
Comment (TomCaserta) [inline src/main/java/net/glowstone/block/state/GlowEnderChest.java:28]: This should return the players enderchest inventory.

**Pull request closed without merging**